### PR TITLE
Add threat protection feature in Connection Tab

### PR DESCRIPTION
**Pull request by renishnair-viper**
When we enable Threat Protection it blocks ads, trackers, and malicious websites once the VPN is connected.

### CODE REVIEW

File: src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs [hunk 1]
```diff
@@ -105,6 +105,12 @@ private void OnSettingsPropertyChanged(object sender, System.ComponentModel.Prop
                 SDK.AllowLocalAdaptersWhenConnected = Properties.Settings.Default.AllowLanInterfaces;
                 Properties.Settings.Default.Save();
             }
+
+            if (e.PropertyName == nameof(Properties.Settings.Default.IsThreatProtectionEnabled))
+            {
+                SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithWLVPNDns : VpnSDK.Enums.DnsFilteringMode.Disabled; 
+                Properties.Settings.Default.Save();
```
Comment (vimalraj-adikesavan) [inline src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs]: "Properties.Settings.Default.Save();" can be avoided as there is no change in the settings properties

File: src/App/Resources/NamedStyles/AppWindow.xaml [hunk 2]
```diff
@@ -124,6 +126,9 @@
                         <Trigger Property="ResizeMode" Value="NoResize">
                             <Setter TargetName="PART_MAXIMIZE_RESTORE" Property="Visibility" Value="Collapsed" />
                         </Trigger>
+                        <Trigger Property="WindowState" Value="Maximized">
+                            <Setter TargetName="MainContentPresenter" Property="Margin" Value="0,0,15,40" />
+                        </Trigger>
```
Comment (vimalraj-adikesavan) [inline src/App/Resources/NamedStyles/AppWindow.xaml]: Does it work in different screen resolutions?

File: src/App/ViewModels/Settings/SplitTunnelingContainerViewModel.cs [hunk 3]
```diff
@@ -29,7 +29,7 @@ public bool IsSplitTunnelingOn
 
         public ISDK SDK { get; }
 
-        public bool CanAddItems => !IsSplitTunnelingOn && !SDK.IsConnected;
+        public bool CanAddItems => IsSplitTunnelingOn;
```
Comment (vimalraj-adikesavan) [inline src/App/ViewModels/Settings/SplitTunnelingContainerViewModel.cs]: Is it allowed to add items when connected to VPN server?

File: src/App/Resources/NamedStyles/AppWindow.xaml (hunk 2 above)
Comment (renishnair-viper) [reply src/App/Resources/NamedStyles/AppWindow.xaml]: Did a minor improvement and it works with all screen resolutions.

File: src/App/ViewModels/Settings/SplitTunnelingContainerViewModel.cs (hunk 3 above)
Comment (renishnair-viper) [reply src/App/ViewModels/Settings/SplitTunnelingContainerViewModel.cs]: Did a change and it is not allowed to add items when connected.

**Review by vimalraj-adikesavan — APPROVED** [review]

**Review by rastapopoulos1 — APPROVED** [review]

**Review by vimalraj-adikesavan — APPROVED** [review]

File: src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs [hunk 4]
```diff
@@ -105,6 +105,11 @@ private void OnSettingsPropertyChanged(object sender, System.ComponentModel.Prop
                 SDK.AllowLocalAdaptersWhenConnected = Properties.Settings.Default.AllowLanInterfaces;
                 Properties.Settings.Default.Save();
             }
+
+            if (e.PropertyName == nameof(Properties.Settings.Default.IsThreatProtectionEnabled))
+            {
+                SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithPartnerDns : VpnSDK.Enums.DnsFilteringMode.Disabled;
```
Comment (rastapopoulos1) [inline src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs]: @renishnair-viper this needs to be 

SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithWLVPNDns: VpnSDK.Enums.DnsFilteringMode.Disabled; so the code is in sync with https://github.com/wlvpn/ConsumerVPN-Windows/pull/14/files#diff-0633ab59162df35a92a0f89f6f2604aa97ad69fddf1359704cd3c1d0d1d33560R104

Comment (renishnair-viper) [reply src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs]: Updated SDK.DnsFilterMode as per suggestion

**Review by rastapopoulos1 — APPROVED** [review]

**Review by vimalraj-adikesavan — APPROVED** [review]

**Pull request merged**